### ISSUE TITLE
numpy should not be referenced in the setup.py

### ISSUE THREAD

**Issue by 12rambau** [gieseke/bfast#43]
it's it the same vein as #37. 

I've corrected some of the bug in my own fork but i'm now seeing another issue. At line https://github.com/diku-dk/bfast/blob/4fd0c1727ec6ac309966d0ecfeef67e3ef02776b/setup.py#L132, the setup.py file is calling numpy. But there are 0 guarantee that numpy will be available when installing the lib. 

This file should only rely on Python built-in modules like distutils or setuptools. 

If not refactored, the lib cannot be installed as in :
```
pip install -r requirements.txt
```

**Comment by mortvest** [comment]
Yes, the setup script in general is a bit of a mess. It needs to be looked at.
